### PR TITLE
Feature/improve time display behaviours

### PR DESCRIPTION
**Pull request by kayhadrin**
Solves the issue when the user set a date and time, cleared the date, and chose another date, and then, the time was cleared too - which is unexpected.

Also show the default time as 00:00 when the initial date (`ngModel`) is `null`.

### CODE REVIEW

File: datetimepicker.js [hunk 1]
```diff
@@ -132,12 +147,32 @@ angular.module('ui.bootstrap.datetimepicker',
...
-          }, function(ngModel) {
+          }, function(newTime) {
             // if a time element is focused, updating its model will cause hours/minutes to be formatted by padding with leading zeros
             if (!element.children()[1].contains(document.activeElement)) {
-              scope.time = new Date(ngModel);
+
+              if (newTime == null || newTime === '') { // if the newTime is not defined
+                if (firstTimeAssign) { // if it's the first time we assign the time value
+                  // create a new default time where the hours, minutes, seconds and milliseconds are set to 0.
+                  newTime = new Date;
```
Comment (zhaber) [inline datetimepicker.js]: newTime = new Date();

File: datetimepicker.js [hunk 2]
```diff
@@ -132,12 +147,32 @@ angular.module('ui.bootstrap.datetimepicker',
...
+                  newTime.setHours(0, 0, 0, 0);
```
Comment (zhaber) [inline datetimepicker.js]: I would default to current time

File: datetimepicker.js (hunk 1 above)
Comment (kayhadrin) [inline datetimepicker.js]: `new Date` is the functionally the same as `new Date()` but I'm happy to change this.

Comment (kayhadrin) [inline datetimepicker.js]: Previously, with `new Date(ngModel)` when `ngModel == null`, you'd get a
date object created from Epoch time at midnight UTC. When a user's computer
is configured with a timezone of, say +11 hours like here in Sydney, you'd
get a default time of 11:00.

The thing to bear in mind is that the date is empty/null atm.
So, to make it clear that the time is somewhat at a "default" state, I
chose to set it to `00:00`.

Should we perhaps propose a new option to set the default time?
e.g.

- current time
- given default time

As a side note, that's also why I've raised another PR to propose to hide
the time inputs when the date is null/empty.

File: datetimepicker.js (hunk 2 above)
Comment (kayhadrin) [inline datetimepicker.js]: Previously, with `new Date(ngModel)` when `ngModel == null`, you'd get a date object created from Epoch time at midnight UTC. When a user's computer is configured with a timezone of, say +11 hours like here in Sydney, you'd get a default time of 11:00. 

The thing to bear in mind is that the date is empty/null atm. So, to make it clear that the time is somewhat at a "default" state, I chose to set it to `00:00`. 

Should we perhaps propose a new option to set the default time? 
e.g. 
- current time 
- given default time 

As a side note, that's also why I've raised another PR to propose to hide the time inputs when the date is null/empty.

**Pull request merged**